src/components/proxy/io_uring_shared.rs

```diff
@@ -660,9 +648,12 @@ impl IoUringLoop {
                                 }
 
                                 let packet = packet.finalize_recv(ret as usize);
-                                if process_packet_tx.blocking_send(packet).is_err() {
-                                    unreachable!("packet process thread has a pending packet");
-                                }
+                                process_packet(
+                                    &mut ctx,
+                                    &process_event_writer,
+                                    packet,
+                                    &mut last_received_at,
+                                );
 
                                 // Queue the wait for the processing of the packet to finish
                                 loop_ctx.push_with_token(
```

Comment:
This can be replaced with the `loop_ctx.enqueue_recv(buffer_pool.clone().alloc());` from below now since the packet is finished so there is no need for an extra loop iteration.